### ISSUE TITLE
Request for a general search api

### ISSUE THREAD

**Issue by MCatherine1994** [bcgov/nr-forest-client-api#152]
**Describe the task**
- We want a general search api, which will allow us to provide either a name (client name) or a forest client number, and the api will return all the possible results based on a wildcard search
- In the return data, we want to get description for the client type code and client status code
- If the user not found, return status code 2xx not 404

For example:
In an input box, we type “CanFor”, we can get all the forest client that has client name includes “CanFor”
And if we type “1234”, we can get all the forest client that has client number includes “1234”

**Acceptance Criteria**
- [ ] A new api that allows us search by either client name or client number, and get a list of possible client with partial match
- [ ] The return client information has whatever the findbyclientnumber api returns, plus a description for the client type code, and a description for the client status code
- [ ] If can't find the user, return status code 2xx not 404

**Comment by MCatherine1994** [comment]
We also would like to request the changes for the response, at least change it when building the search API (if not fix the existing one), based on [this](https://medium.com/codex/should-apis-use-404-as-an-error-code-82e41ac96ffe#:~:text=This%20means%20that%20API%20designers,error%20codes%20for%20missing%20data.) design pattern:

"query APIs should always return 200 OK even if zero rows are found"

"If the status code is between 200–299, treat the data received in the respond body as the result. The results may include nulls or be an empty array. In any case, status codes in this range mean that the results are valid and the code worked as expected.

If the status code is between 400–499, a problem exists and it is the responsibility of the developer calling the API to fix it. They could have expired credentials, a malformed query, or be calling the wrong URL.

If the status code is between 500–599, a bug exists and the developer of the API must fix it. Any service reporting a 500 error should generally update their status page if the outage is not temporary."


In other words, if we do a search / query and there are no results, this should be a HTTP 2xx status code. If we call a URL that doesn't exist (e.g. /saerch instead of /search, then we should get a 404. Currently we get 404 when searching for a forest client that doesn't exist.
